### PR TITLE
Refactor arm_dyncom_interpreter into several files (take 2)

### PR DESCRIPTION
**Pull request by archshift**
See #1893 for take 1.

I've split this into several comments (please review them individually) so that it's no longer one monolithic change.

I've also tested this, and fixed the issue that users were previously happening ( #1895 ). Specifically, the `top` variable of the instruction buffer wasn't being updated after I changed the `AllocBuffer` function, but with this PR it is correctly changed.

### CODE REVIEW

**Comment by archshift** [comment]
Note to self: 9da0e9f8b314682896ccc368ec658c1ff8fd2a9d doesn't build, so I need to backport the fix in the next commit
- [ ] Done

File: src/core/arm/dyncom/arm_dyncom_trans.cpp [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+#include "core/arm/dyncom/arm_dyncom_interpreter.h"
+#include "core/arm/dyncom/arm_dyncom_trans.h"
+#include "core/arm/skyeye_common/armstate.h"
+#include "core/arm/skyeye_common/armsupp.h"
+#include "core/arm/skyeye_common/vfp/vfp.h"
+
+char trans_cache_buf[TRANS_CACHE_SIZE];
+size_t trans_cache_buf_top = 0;
+
+static FORCE_INLINE void* AllocBuffer(size_t size) {
```
Comment (lioncash) [inline src/core/arm/dyncom/arm_dyncom_trans.cpp]: This doesn't need to be force inlined.

**Comment by archshift** [comment]
Addressed. Anything else?

**Comment by bunnei** [comment]
LGTM

**Pull request merged**